src/tsp-client.ts

```diff
@@ -147,6 +149,13 @@ export class TspClient {
             const stringMsg = typeof data === 'string' ? data : dec.write(data);
             this.tsserverLogger.error(stringMsg);
         });
+        return true;
+    }
+
+    shutdown(): void {
```

Comment:
Doesn't look like you ended up using this.

Reply:
Yes. Though I wonder now whether I should restore the `process.on('exit', ...)` handler and call the `shutdown`. As far as my testing shows the child tsserver process is closed automatically but not sure if that's gonna be the case in all situations and on all platforms...

Reply:
Yeah, it definitely depends on the OS and on Linux's pid1 implementation. I'd say it's safer if you did add it.